### PR TITLE
Phoenix: from chapter 4 to 6

### PR DESCRIPTION
**Pull request by Vretamoza**

### CODE REVIEW

File: hello/lib/hello_web/plugs/locale.ex [hunk 1]
```diff
@@ -0,0 +1,15 @@
+defmodule HelloWeb.Plugs.Locale do
```
Comment (juanhurtado10) [inline hello/lib/hello_web/plugs/locale.ex:1]: @Vretamoza what does a plug do?

File: hello/lib/hello_web/router.ex [hunk 2]
```diff
@@ -8,6 +8,7 @@ defmodule HelloWeb.Router do
     plug :put_root_layout, {HelloWeb.LayoutView, :root}
     plug :protect_from_forgery
     plug :put_secure_browser_headers
+    plug HelloWeb.Plugs.Locale, "en"
   end
 
   pipeline :api do
```
Comment (juanhurtado10) [inline hello/lib/hello_web/router.ex:14]: @Vretamoza what a pipeline is?

File: hello/lib/hello_web/router.ex [hunk 3]
```diff
@@ -18,7 +19,10 @@ defmodule HelloWeb.Router do
     pipe_through :browser
 
     get "/", PageController, :index
+
+    resources "/users", UserController
```
Comment (juanhurtado10) [inline hello/lib/hello_web/router.ex:23]: @Vretamoza what is the main difference between resources and get/post?

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: hello/lib/hello_web/plugs/locale.ex (hunk 1 above)
Comment (Vretamoza) [reply hello/lib/hello_web/plugs/locale.ex:1]: A plug is a specification that helps to build endpoints. It receives a conn, modifies it and return it.

File: hello/lib/hello_web/router.ex (hunk 2 above)
Comment (Vretamoza) [reply hello/lib/hello_web/router.ex:14]: A pipeline is a set of plugs which are introduced in a scope in order to pass all the requests in that scope through all of those plugs.

File: hello/lib/hello_web/router.ex (hunk 3 above)
Comment (Vretamoza) [reply hello/lib/hello_web/router.ex:23]: When we use the resources verb, it creates all the standard verbs, paths and controller actions for that request, when we use the get/post verb, we just create that individual request to a single path and invoke a single controller action.

**Review by juanhurtado10 — APPROVED** [review]

**Pull request merged**